lib/Doctrine/DBAL/Connection.php

```diff
@@ -1034,9 +1033,9 @@ public function query()
      *
      * This method supports PDO binding types as well as DBAL mapping types.
      *
-     * @param string         $query  The SQL query.
-     * @param mixed[]        $params The query parameters.
-     * @param int[]|string[] $types  The parameter types.
+     * @param string                 $query  The SQL query.
+     * @param mixed[]                $params The query parameters.
+     * @param array<int|string|null> $types  The parameter types.
```

Comment:
Should `$params` and `$types` have the same "container" type since they both describe parameters? I.e. `mixed[]` → `array<mixed>`.